### PR TITLE
Make test_analysis PyTest

### PR DESCRIPTION
**Pull request by trexfeathers**
`ruff check --select=PT` compliant

### CODE REVIEW

File: lib/iris/tests/test_analysis_calculus.py [hunk 1]
```diff
@@ -733,19 +760,22 @@ def test_contrived_spherical_curl2(self):
         np.testing.assert_array_almost_equal(
```
Comment (ESadek-MO) [inline lib/iris/tests/test_analysis_calculus.py:760]: I've been tending to convert these calls to _shared_utils for sake of consistency.

**Review by ESadek-MO — CHANGES_REQUESTED** [review]
Only thing I could see was the use of np.testing functions, otherwise LGTM!

File: lib/iris/tests/test_analysis_calculus.py (hunk 1 above)
Comment (trexfeathers) [reply lib/iris/tests/test_analysis_calculus.py]: bcaca28a6a14cf62f30e034db5b6aa02e1029303

**Review by ESadek-MO — APPROVED** [review]
thanks @trexfeathers

**Pull request merged**